api/README.md

```diff
@@ -124,6 +123,39 @@ If everything was setup correctly, you should see a response that resembles the
 
 - [Swagger](https://swagger.io/) XML Comments have been enabled to provide better API Documentation.  This means that warnings will be generated for public undocumented public types and members.  By default, this project disables warnings.  Documentation to enable warnings can be found [here](https://docs.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-3.1&tabs=visual-studio-code)
 
+# Testing
+
+## Running unit tests
+Unit tests validate the individual components of the API
+```bash
+dotnet test api/tests/LinkyLink.Tests/LinkyLink.Tests.csproj
+```
+
+## Running Integration tests
+Integration tests validate the APIs against a running system.
+
+Before running integration tests you must update the settings within the `appsettings.json` of the integration test project or set environment variables for the following:
+
+### Set Environment Variables
+```bash
+# The base address of the environment to test
+export INTTEST_AzureAdB2C__BaseAddress= 
+# The client ID of your test application within Azure B2C
+export INTTEST_AzureAdB2C__ClientId= 
+# The Azure B2C authority endpoint
+export INTTEST_AzureAdB2C__Authority= 
+# The scope to set on your access token
+export INTTEST_AzureAdB2C__Scope= 
+# The username of your test user account
+export INTTEST_AzureAdB2C__Username= 
+# The password of your test user account
+export INTTEST_AzureAdB2C__Password= 
+```
```

Comment:
Some questions:
1) Out of curiosity, why double underscores before the attribute name?
2) Would it be possible to describe where they could get these values?
3) what is the test user account here?
4) I might be over-thinking it, but is there a better way than to pass the password here?

Reply:
The double underscore is a convention used by the .net core configuration provider.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#environment-variables-configuration-provider

I can add links to learn more about Azure B2C configuration.  I wouldn't want to duplicate existing documentation.

In order to test the API as an authenticated user we need to authenticate.  The integration tests use a OAuth password grant which allows us to use local user accounts associated to the Azure B2C tenant (not social accounts).

Without authenticating we also would not be able to test many of the endpoints for getting links by user, deleting links, etc.

Reply:
Updated readme with some more information.

Reply:
Got it, makes sense and thanks for the response, insightful as always 😄